### PR TITLE
A bunch of grammar and spelling corrections.

### PR DESCRIPTION
**Pull request by polarbub**
I fixed lots of formatting, spelling, and grammar of the articles. I think I also changed a bit of css for code blocks. I did all of the articles in the `Blocks` - `Entities` folders, all of the main subject pages, and a few other random ones. The TNT Duping article seemed factually wrong so I couldn't correct it.

### CODE REVIEW

File: Blocks/MovingBlock36.md [hunk 1]
```diff
@@ -1,38 +1,39 @@
 ---
-title: Moving piston block 36
+title: Moving block 36
 description: Mechanics around the moving piston block
 ---
 
-# moving piston block 36
+# Moving piston block 36
```
Comment (JoakimThorsen) [inline Blocks/MovingBlock36.md]: does not have the same change as the title. is that an oversight?

File: Blocks/MovingBlock36.md [hunk 2]
```diff
@@ -1,38 +1,39 @@
 ---
-title: Moving piston block 36
+title: Moving block 36
```
Comment (JoakimThorsen) [inline Blocks/MovingBlock36.md]: perhaps `Moving block (block 36)` is a better title?

File: Blocks/MovingBlock36.md [hunk 3]
```diff
@@ -1,38 +1,39 @@
...
 description: Mechanics around the moving piston block
 ---
 
-# moving piston block 36
+# Moving piston block 36
+
+Moving_Piston is commonly called a B36 or Block 36. This is because, before The Flattening<!--FIX: What's that?-->, it had the block ID 36. B36 is created by pistons when they push a block. The piston head also converts itself into a B36. B36 is non-solid<!--FIX:Seems to contradict the part where it says it has a hitbox--> and invisible, although it does have a hitbox.
 
-Moving_Piston is commonly called a B36 or Block 36 in the technical community. This is because, before The Flattening, it had the block ID 36.
-B36 is created through pistons. When a piston pushes a set of blocks, it transforms all blocks in that set into B36 while they move. The piston head also converts itself into a B36. It is non-solid and invisible, although it does have a hitbox.
```
Comment (JoakimThorsen) [inline Blocks/MovingBlock36.md:9]: the flattening could possibly link to the mc wiki for now? https://minecraft.fandom.com/wiki/Java_Edition_1.13/Flattening

"non-solid" is an old and vague term, it's probably meant in the sense of being non-opaque and non-conductive.

File: Blocks/MovingBlock36.md [hunk 4]
```diff
@@ -1,38 +1,39 @@
...
 Special properties that B36 has which might be useful include:
-- Preventing falling_blocks from aging (falling blocks won't break while inside a B36);
+- Preventing falling blocks from aging (falling blocks won't break while inside a B36);
 - Having the Block Tags: dragon_immune, wither_immune;
-- Entities can see through it but cannot path through it;
-- Being destroyed only through an explosion;
-- Being unreplaceable by the player
+- Entities can see through it but cannot pathfind through it;
+- It can only be destroyed through an explosion;
+- Being unreplaceable<!--What does this mean?--> by the player
```
Comment (JoakimThorsen) [inline Blocks/MovingBlock36.md]: cannot be mined or replaced by right-clicking with another block item (like tallgrass can, for example)

File: Blocks/MovingBlock36.md [hunk 5]
```diff
@@ -1,38 +1,39 @@
...
-### Hitbox
-Like most blocks, the B36 has a hitbox. A B36 hitbox is very interesting as it can be offset from where the B36 is actually located. This can be seen if you do the following:
+## Hitbox
+Like most blocks, the B36 has a hitbox. A B36 hitbox is very interesting as it can be offset from where the B36 is actually located. This can be seen by doing the following:
 1. Go to the world border
 2. Place a piston facing into the world border, with water on the other side.
-3. Extend the piston
-4. Without unpowering the piston, break the piston
+3. Power the piston to extend it
+4. Without unpowering the piston, break the it
```
Comment (JoakimThorsen) [inline Blocks/MovingBlock36.md]: "break it"?

File: BugsAndExploits/LightSupression.md [hunk 6]
```diff
@@ -1,22 +1,20 @@
...
 
-# light update suppression
+# Light Suppression
 
 Light update suppression is the delaying of light updates and then stopping the server to effectively delete them.
 
-## Technical explaination
-When a light update is produced via a block being moved, placed or broken it is placed in a queue for the light thread to process. Then when the light thread gets to the update it is processed and the light level is changed. How light update suppression works is it overloads that queue; delaying the processing of light updates. Then when the server is stopped the unprocessed light updates are never processed and the light level is wrong.
+## Technical explanation
+When a block being moved, placed, or broken a light update is produced. These light updates are placed in a queue for the light thread to process. Then when the light thread gets to that update it processes it and the light level is changed in the world. Light Suppression works by filling that queue with other light updates; delaying the processing of them. When the server is stopped any unprocessed light updates are deleted and thereby not processed, so the light level is wrong.
```
Comment (JoakimThorsen) [inline BugsAndExploits/LightSupression.md]: > When a block being moved, placed, or broken, a light update is produced. These light updates are placed in a queue for the light thread to process. When the light thread gets to that update it processes it and the light level is changed in the world. The light thread and the main game thread are asynchronous; this means that if one thread is slowing down, the other can still keep going. Light Suppression works by filling that queue with too many light updates, which will cause the light thread to lag behind the main game thread, meaning that any new light updates queued from the main thread will only be processed if the light thread gets a chance to "catch up". When the server is stopped, any unprocessed light updates are deleted and thereby not processed, so the light level will stay in this incorrect state.

thoughts?

File: BugsAndExploits/LightSupression.md [hunk 7]
```diff
@@ -1,22 +1,20 @@
...
-After aproximatly 20k light updates in the queue the machine is stopped. Normally this length of queue is around 3 seconds to process a light update. Any longer and the server will slow down to let the light thread catch up. 
-A light update that is to be delted is produced.
-The server is stopped either through the `/stop`
- command or crashing with an update suppressor. Results with `/stop`
- can be inconsistant so `/save-all flush` can be ran before to fix that. 
+1. A light suppression machine is run. This typically consists of flashing redstone lamps or, in older light update suppressors, moving blocks. Light updates from light sources typically take longer to process.
+2. After approximately 20k light updates are in the queue the machine is stopped. Normally the length of queue is around 3 seconds because if it were any longer the server will slow down to let the light thread catch up. 
+3. A light update that is to be deleted is produced.
+4. The server is stopped either through the `/stop` command or crashing with an update suppressor. Results with `/stop` can be inconsistent so `/save-all flush` can be ran before to fix that.
```
Comment (JoakimThorsen) [inline BugsAndExploits/LightSupression.md:19]: can someone verify point 2? does not sound correct to me.
point 3 is also phrased quite oddly

File: BugsAndExploits/TntDuping.md [hunk 8]
```diff
@@ -30,3 +25,9 @@ These specific blocks are able to create a normal update after the list of block
 That needs to happen before the TNT block is removed: before a block behind the tnt replaces it. if there is no blocks, it can be after the moving TNT was created.
 This update can also be propagated through other blocks, as long as it is instant (for example using rails, dust, noteblocks (1.13+), or TNT itself), which allows for up to 11 TNT to be duplicated using a single piston push, the 12th block being for updating.
 The tnt moving block is created from the moved block list regardless of the original tnt being ignited or not, and this means that the TNT is preserved in block form.
+
+## Mojang's Opposition
+Mojang historically opposes TNT Duping as they consider it an exploit, and they have to patch it four times, but in the 1.16 snapshots they declared they will leave it in the game for the time being, due to a lack of replacement for what TNT duping currently allows: [https://www.reddit.com/r/Minecraft/comments/fkt9jf/anchor_yourself_to_the_nether_snapshot_20w12a_is/fkv33q3](https://www.reddit.com/r/Minecraft/comments/fkt9jf/anchor_yourself_to_the_nether_snapshot_20w12a_is/fkv33q3)
```
Comment (JoakimThorsen) [inline BugsAndExploits/TntDuping.md]: "have tried to"

File: BugsAndExploits/ZeroTickFarms.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Zero tick farms
-description: How a zero tick farms work
+title: Zero Tick Farms
+description: How a Zero Tick Farms work
```
Comment (JoakimThorsen) [inline BugsAndExploits/ZeroTickFarms.md]: is this singular or plural?

**Review by JoakimThorsen — COMMENTED** [review]
I like a lot of the reformulations and edits, especially the "information pertaining to"-changes. I appreciate the effort put in.

I've listed some proposed changes and tweaks, plus a rewrite of the Light Suppression>Technical Explanation section (couldn't figure out how to "suggest a change", sorry about that)

File: BugsAndExploits/ZeroTickFarms.md (hunk 9 above)
Comment (Dainternetdude) [reply BugsAndExploits/ZeroTickFarms.md]: yeah i agree and this is nitpicking but the title should probably be something like "Zero-Tick Farming"

File: BugsAndExploits/TntDuping.md (hunk 8 above)
Comment (Dainternetdude) [reply BugsAndExploits/TntDuping.md]: I would add this for more context: https://www.reddit.com/r/Minecraft/comments/e5gx0s/comment/f9q2eih/

**Comment by polarbub** [comment]
If I click resolve conversation does it accept the suggested changes, or does it just mark it as done?

**Comment by JoakimThorsen** [comment]
@polarbub they're just comments, not direct change suggestions, so it will just mark them as resolved

File: BugsAndExploits/LightSupression.md (hunk 7 above)
Comment (polarbub) [reply BugsAndExploits/LightSupression.md:19]: I have experienced point 2 in 1.17+. I have no ideas how to phrase point three any better. Any ideas?

File: BugsAndExploits/LightSupression.md (hunk 6 above)
Comment (polarbub) [reply BugsAndExploits/LightSupression.md]: Looks good, I added it.

File: BugsAndExploits/TntDuping.md (hunk 8 above)
Comment (polarbub) [reply BugsAndExploits/TntDuping.md]: Done

File: BugsAndExploits/ZeroTickFarms.md (hunk 9 above)
Comment (polarbub) [reply BugsAndExploits/ZeroTickFarms.md]: Done

File: Blocks/MovingBlock36.md (hunk 5 above)
Comment (polarbub) [reply Blocks/MovingBlock36.md]: Done

File: Blocks/MovingBlock36.md (hunk 4 above)
Comment (polarbub) [reply Blocks/MovingBlock36.md]: I added that for context

File: Blocks/MovingBlock36.md (hunk 3 above)
Comment (polarbub) [reply Blocks/MovingBlock36.md:9]: Linked

File: Blocks/MovingBlock36.md (hunk 2 above)
Comment (polarbub) [reply Blocks/MovingBlock36.md]: I added that.

File: Blocks/MovingBlock36.md (hunk 1 above)
Comment (polarbub) [reply Blocks/MovingBlock36.md]: Fixed

**Comment by polarbub** [comment]
I think I fixed all the issues you guys found. Does it look good now?

File: BugsAndExploits/ZeroTickFarms.md [hunk 10]
```diff
@@ -1,12 +1,12 @@
 ---
-title: Zero Tick Farms
-description: How a Zero Tick Farms work
+title: Zero Tick Farming
+description: How a Zero Tick Farming work
```
Comment (JoakimThorsen) [inline BugsAndExploits/ZeroTickFarms.md]: title: Zero-Tick Farming
description: How Zero-Tick Farming works

**Comment by JoakimThorsen** [comment]
that^ is my only nitpick, looks good otherwise :)

File: BugsAndExploits/ZeroTickFarms.md (hunk 10 above)
Comment (polarbub) [reply BugsAndExploits/ZeroTickFarms.md]: Fixed

**Pull request merged**